lib/analyze_hoc_activity_helper.rb

```diff
@@ -9,23 +9,78 @@
 DASHBOARD_REPORTING_DB_READONLY = sequel_connect(CDO.dashboard_reporting_db_reader, CDO.dashboard_reporting_db_reader)
 PEGASUS_REPORTING_DB_READONLY = sequel_connect(CDO.pegasus_reporting_db_reader, CDO.pegasus_reporting_db_reader)
 
+WEIGHTED_COUNT = "SUM(" \
+  " IF(session REGEXP '^_.*_'," +
+  # Parses the session weight 'xyz' from the session '_xyz_abcdefghijk'.
+  "   SUBSTRING(SUBSTRING_INDEX(session, '_', 2) FROM 2)," +
+  # The session does not specify the session weight, default to one.
+  "   1)" \
+  " ) AS count"
+
+def generate_from_where(start_date, end_date: nil, tutorial: true, tutorial_pixel: true, finished: false)
```

Comment:
nit: Please document that `start_date` is inclusive and `end_date` is exclusive.

Better, document the meaning of all parameters in a comment, as none are crystal clear.